### ISSUE TITLE
SchedulerPro v5 loadInlineData doesn't work as async

### ISSUE THREAD

**Issue by majewski-merix** [bryntum/support#4452]
We are upgrading SchedulerPro from 4.3.6 to 5.0.1 and using `loadInlineData` we don't have all the proper data right after initialization.

Here is a console output for v4 with `window.DEBUG = true`
<img width="270" alt="image" src="https://user-images.githubusercontent.com/21220597/161524546-f6769aed-c599-4718-90b2-7927cc92e552.png">
As you can see we have 2 events with assignments.

But in v5 we don't have that assignments on event change `dataset` and event is populated before data is set
<img width="297" alt="image" src="https://user-images.githubusercontent.com/21220597/161525066-83c5fe1f-454a-4d98-ab3a-5dad712494aa.png">

After using `setTImeout` we have all the proper data.

`loadInlineData` function is called with `await` in both cases.

**Comment by chuckn0rris** [comment]
I tried to call `schedulerPro.loadInlineData(data)` right after schedulerPro initialization (next line of the code), and I don't see any problems, all works as expected. 

Please provide a test case how to reproduce this problem.

**Comment by majewski-merix** [comment]
Tried to reproduce it on 1 of the example pages and couldn't do that. We are defining stores in the config's project and when I wanted to duplicate it in the example got an error `onCommitAsync is not a function`. If I don't use that example code everything is working but locally without custom stores it's still not working properly. 
Can't use the trial version locally because it expired already.

Here I'll try to explain what we have.

We are using React and remote data. After fetching data, we call
```js
await refs.current.scheduler.instance.project.loadInlineData({
    eventsData: eventsList,
    resourcesData: resources,
    assignmentsData: tasksAssignments,
    resourceTimeRangesData: resourceTimeRanges,
});
```

and in another component where we have a list of unscheduled items (items that don't have any assignments) we use `ventStore.on("change", event => {});` where we relly on action name (dataset, add, update). And when we want to filter those events we don't have proper data. Using `setTimeout` gives us proper values but it's not how we would like to handle that since it may create other bugs and side effects.

We have 1 custom store which extends from EventStore, others are regular. This doesn't matter since even without `project` config it's not working.

Also, when I run `await scheduler.project.loadInlineData(inlineData)` locally I've got a return `{ rejectedWith: undefined }` but the same call on the example page gives me a result `undefined` which looks like there are differences in the versions.

In the project, we use the npm package if it matters.

**Comment by majewski-merix** [comment]
Tried also to update data by `loadInlineData` and the events jump from old date to new date and back to old, here is a recording. It's not reproducible on any example app from https://bryntum.com/examples/scheduler-pro/

It's only on the date-time change, changing assignments works
 
https://user-images.githubusercontent.com/21220597/162215949-b0cabb95-4275-46ff-a239-c362fa5a2263.mov

The code for the update 
```
setTimeout(async () => {
    await scheduler.project.loadInlineData({
        eventsData: [
            ...events.slice(0, 1),
            {
                ...events[1],
                startDate  : new Date(2017, 0, 1, 16),
                endDate    : new Date(2017, 0, 1, 17),
            },
            ...events.slice(2),
        ],
        resourcesData: resources,
        assignmentsData: assignments,
    })
}, 10000)

**Comment by chuckn0rris** [comment]
We don't see any problems in online examples using that code. That sounds like some race-condition problem, but unfortunately we can't reproduce that. Please check if you have data binding with your framework and try to remove it and check if it still reproducible. Try to simplify your app by disabling code part by part to see what config is causing that.

**Comment by majewski-merix** [comment]
@chuckn0rris does this code works for you?
```js
import { Scheduler, EventStore, AssignmentStore, ResourceStore } from '../../build/schedulerpro.module.js';
import shared from '../_shared/shared.module.js';

//region Data

const
    resources   = [
        { id : 'r1', name : 'Celia', city : 'Barcelona' },
        { id : 'r2', name : 'Lee', city : 'London' },
        { id : 'r3', name : 'Macy', city : 'New York' },
        { id : 'r4', name : 'Madison', city : 'Barcelona' },
        { id : 'r5', name : 'Rob', city : 'Rome' },
        { id : 'r6', name : 'Dave', city : 'Barcelona' },
        { id : 'r7', name : 'Dan', city : 'London' },
        { id : 'r8', name : 'George', city : 'New York' },
        { id : 'r9', name : 'Gloria', city : 'Rome' },
        { id : 'r10', name : 'Henrik', city : 'London' }
    ],
    events      = [
        {
            id        : 1,
            startDate : new Date(2017, 0, 1, 10),
            endDate   : new Date(2017, 0, 1, 12),
            name      : 'Multi assigned',
            iconCls   : 'b-fa b-fa-users'
        },
        {
            id         : 2,
            startDate  : new Date(2017, 0, 1, 13),
            endDate    : new Date(2017, 0, 1, 15),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'indigo'
        },
        {
            id         : 3,
            startDate  : new Date(2017, 0, 1, 8),
            endDate    : new Date(2017, 0, 1, 11),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'cyan'
        },
        {
            id         : 4,
            startDate  : new Date(2017, 0, 1, 10),
            endDate    : new Date(2017, 0, 1, 13),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'blue'
        },
        {
            id         : 5,
            startDate  : new Date(2017, 0, 1, 13),
            endDate    : new Date(2017, 0, 1, 15),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'violet'
        }
    ],
    assignments = [
        { id : 1, resourceId : 'r1', eventId : 1 },
        { id : 2, resourceId : 'r2', eventId : 1 },
        { id : 3, resourceId : 'r8', eventId : 1 },
        { id : 4, resourceId : 'r3', eventId : 2 },
        { id : 5, resourceId : 'r4', eventId : 3 },
        { id : 6, resourceId : 'r5', eventId : 4 },
        { id : 7, resourceId : 'r6', eventId : 5 }
    ];

//endregion

const scheduler = new Scheduler({
    appendTo : 'container',

    startDate         : new Date(2017, 0, 1, 6),
    endDate           : new Date(2017, 0, 1, 20),
    viewPreset        : 'hourAndDay',
    eventStyle        : 'border',
    resourceImagePath : '../_shared/images/users/',

    columns : [
        { type : 'resourceInfo', text : 'Name', field : 'name', width : 130 },
        { text : 'City', field : 'city', width : 90 }
    ],

    project: {
        repopulateOnDataset: false,
        eventStore: new EventStore(),
        resourceStore: new ResourceStore(),
        assignmentStore: new AssignmentStore(),
    },
});

const loadData = async () => {
    await scheduler.project.loadInlineData({
        eventsData: events,
        resourcesData: resources,
        assignmentsData: assignments,
    });
}

loadData();
```

in this example https://bryntum.com/examples/examples-scheduler/multiassign/

Because I've got this errors in the console
<img width="600" alt="image" src="https://user-images.githubusercontent.com/21220597/162676578-808904c2-6094-463f-be12-70946d9edc4b.png">


And this is simplified example what we are doing.

**Comment by chuckn0rris** [comment]
The problem you highlighted here is because you used Scheduler, not SchedulerPro, but imported EventStore/AssignmentStore/ResourceStore classes for SchedulerPro. 
If you need to import these stores from SchedulerPro package, but for Scheduler, please use 'Scheduler' prefix:

```
import { Scheduler, SchedulerEventStore, SchedulerAssignmentStore, SchedulerResourceStore } from '../../build/schedulerpro.module.js';

// ...
    project: {
        repopulateOnDataset: false,
        eventStore: new SchedulerEventStore(),
        resourceStore: new SchedulerResourceStore(),
        assignmentStore: new SchedulerAssignmentStore(),
    },
```

I tested locally and with online examples, it works well.
We are noticed that this moment has not been well described in our docs and will fix it soon. 

Try to check your imports in your initial app and let us know if it's related to the problem you described in the beginning.

**Comment by majewski-merix** [comment]
Thanks for the tip, I didn't saw that because I was 100% sure that this example will use SchedulerPro since I'm looking at pro examples.
Anyway, I was able to reproduce our issue. Here is updated code
```js
import { SchedulerPro, EventStore, AssignmentStore, ResourceStore } from '../../build/schedulerpro.module.js';
import shared from '../_shared/shared.module.js';

//region Data

const
    resources   = [
        { id : 'r1', name : 'Celia', city : 'Barcelona' },
        { id : 'r2', name : 'Lee', city : 'London' },
        { id : 'r3', name : 'Macy', city : 'New York' },
        { id : 'r4', name : 'Madison', city : 'Barcelona' },
        { id : 'r5', name : 'Rob', city : 'Rome' },
        { id : 'r6', name : 'Dave', city : 'Barcelona' },
        { id : 'r7', name : 'Dan', city : 'London' },
        { id : 'r8', name : 'George', city : 'New York' },
        { id : 'r9', name : 'Gloria', city : 'Rome' },
        { id : 'r10', name : 'Henrik', city : 'London' }
    ],
    events      = [
        {
            id        : 1,
            startDate : new Date(2017, 0, 1, 10),
            endDate   : new Date(2017, 0, 1, 12),
            name      : 'Multi assigned',
            iconCls   : 'b-fa b-fa-users'
        },
        {
            id         : 2,
            startDate  : new Date(2017, 0, 1, 13),
            endDate    : new Date(2017, 0, 1, 15),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'indigo'
        },
        {
            id         : 3,
            startDate  : new Date(2017, 0, 1, 8),
            endDate    : new Date(2017, 0, 1, 11),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'cyan'
        },
        {
            id         : 4,
            startDate  : new Date(2017, 0, 1, 10),
            endDate    : new Date(2017, 0, 1, 13),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'blue'
        },
        {
            id         : 5,
            startDate  : new Date(2017, 0, 1, 13),
            endDate    : new Date(2017, 0, 1, 15),
            name       : 'Single assigned',
            iconCls    : 'b-fa b-fa-user',
            eventColor : 'violet'
        }
    ],
    assignments = [
        { id : 1, resourceId : 'r1', eventId : 1 },
        { id : 2, resourceId : 'r2', eventId : 1 },
        { id : 3, resourceId : 'r8', eventId : 1 },
        { id : 4, resourceId : 'r3', eventId : 2 },
        { id : 5, resourceId : 'r4', eventId : 3 },
        { id : 6, resourceId : 'r5', eventId : 4 },
        { id : 7, resourceId : 'r6', eventId : 5 }
    ];

//endregion

const eventStore = new EventStore();

eventStore.on("change", event => {
    if (event.action === "dataset") {
        event.records.forEach(
            eventRecord => {
                console.log('eventRecord.assignments.length: ', eventRecord.assignments.length);
            }
        );
        setTimeout(() => {
            console.log("after timeout");
            event.records.forEach(
                eventRecord => {
                    console.log('eventRecord.assignments.length: ', eventRecord.assignments.length);
                }
            );
        });
    }
});

const scheduler = new SchedulerPro({
    appendTo : 'container',

    startDate         : new Date(2017, 0, 1, 6),
    endDate           : new Date(2017, 0, 1, 20),
    viewPreset        : 'hourAndDay',
    eventStyle        : 'border',
    resourceImagePath : '../_shared/images/users/',

    columns : [
        { type : 'resourceInfo', text : 'Name', field : 'name', width : 130 },
        { text : 'City', field : 'city', width : 90 }
    ],

    project: {
        repopulateOnDataset: false,
        eventStore,
        resourceStore: new ResourceStore(),
        assignmentStore: new AssignmentStore(),
    },
});

const loadData = async () => {
    await scheduler.project.loadInlineData({
        eventsData: events,
        resourcesData: resources,
        assignmentsData: assignments,
    });
}

loadData();
```
In `dataset` event on the first event records forEach we have assignments length equal to 0 but the same forEach but in timeout give us proper values. Here is also a console output
<img width="270" alt="image" src="https://user-images.githubusercontent.com/21220597/162695035-d424cfc1-2343-4aae-b7cf-b0983272bde6.png">

**Comment by majewski-merix** [comment]
Used your newest version (5.0.3) and new inline data example and this is still not working as it should

```js
import { SchedulerPro } from '../../build/schedulerpro.module.js?458603';
import shared from '../_shared/shared.module.js?458603';
export const data = {

    resources : [
        { id : 1, name : 'Truck 1', iconCls : 'b-fa b-fa-truck', image : false },
        { id : 2, name : 'Truck 2', iconCls : 'b-fa b-fa-truck', image : false },
        { id : 3, name : 'Truck 3', iconCls : 'b-fa b-fa-truck', image : false },
        { id : 4, name : 'Train 1', iconCls : 'b-fa b-fa-train', image : false },
        { id : 5, name : 'Train 2', iconCls : 'b-fa b-fa-train', image : false },
        { id : 6, name : 'Pickup 1', iconCls : 'b-fa b-fa-truck-pickup', image : false },
        { id : 7, name : 'Pickup 2', iconCls : 'b-fa b-fa-truck-pickup', image : false },
        { id : 8, name : 'Pickup 3', iconCls : 'b-fa b-fa-truck-pickup', image : false },
        { id : 9, name : 'Plane 1', iconCls : 'b-fa b-fa-plane', image : false },
        { id : 10, name : 'Plane 2', iconCls : 'b-fa b-fa-plane', image : false },
        { id : 11, name : 'Helicopter 1', iconCls : 'b-fa b-fa-helicopter', image : false },
        { id : 12, name : 'Rocket 1', iconCls : 'b-fa b-fa-rocket', image : false },
        { id : 13, name : 'Rocket 2', iconCls : 'b-fa b-fa-rocket', image : false },
        { id : 14, name : 'Horse', iconCls : 'b-fa b-fa-horse', image : false }
    ],

    events : [
        {
            id         : 1,
            name       : 'Arrive',
            startDate  : '2022-03-23T03:00',
            endDate    : '2022-03-23T05:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 2,
            name       : 'Unload',
            startDate  : '2022-03-23T06:00',
            endDate    : '2022-03-23T09:00',
            iconCls    : 'b-fa b-fa-snowplow',
            eventColor : 'orange'
        },
        {
            id         : 3,
            name       : 'Load',
            startDate  : '2022-03-23T10:00',
            endDate    : '2022-03-23T12:00',
            iconCls    : 'b-fa b-fa-truck-loading',
            eventColor : 'green'
        },
        {
            id         : 4,
            name       : 'Depart',
            startDate  : '2022-03-23T13:00',
            endDate    : '2022-03-23T14:30',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 5,
            name       : 'Arrive 2',
            startDate  : '2022-03-23T07:00',
            endDate    : '2022-03-23T09:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 6,
            name       : 'Unload 2',
            startDate  : '2022-03-23T10:00',
            endDate    : '2022-03-23T13:30',
            iconCls    : 'b-fa b-fa-snowplow',
            eventColor : 'orange'
        },
        {
            id         : 7,
            name       : 'Load 2',
            startDate  : '2022-03-23T14:30',
            endDate    : '2022-03-23T16:00',
            iconCls    : 'b-fa b-fa-truck-loading',
            eventColor : 'green'
        },
        {
            id         : 8,
            name       : 'Depart 2',
            startDate  : '2022-03-23T17:00',
            endDate    : '2022-03-23T19:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 9,
            name       : 'Arrive 3',
            startDate  : '2022-03-23T01:00',
            endDate    : '2022-03-23T05:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 10,
            name       : 'Unload 3',
            startDate  : '2022-03-23T06:00',
            endDate    : '2022-03-23T09:00',
            iconCls    : 'b-fa b-fa-hand-paper',
            eventColor : 'orange'
        },
        {
            id         : 11,
            name       : 'Load 3',
            startDate  : '2022-03-23T10:00',
            endDate    : '2022-03-23T15:00',
            iconCls    : 'b-fa b-fa-user-astronaut',
            eventColor : 'green'
        },
        {
            id         : 12,
            name       : 'Depart 3',
            startDate  : '2022-03-23T16:00',
            endDate    : '2022-03-23T21:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 13,
            name       : 'Arrive 4',
            startDate  : '2022-03-23T05:00',
            endDate    : '2022-03-23T09:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 14,
            name       : 'Reload 4',
            startDate  : '2022-03-23T11:00',
            endDate    : '2022-03-23T16:00',
            iconCls    : 'b-fa b-fa-snowplow',
            eventColor : 'orange'
        },
        {
            id         : 15,
            name       : 'Depart 4',
            startDate  : '2022-03-23T18:00',
            endDate    : '2022-03-23T22:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 16,
            name       : 'Yearly maintenance',
            startDate  : '2022-03-23T05:00',
            endDate    : '2022-03-23T13:00',
            iconCls    : 'b-fa b-fa-wrench',
            eventColor : 'gray'
        },
        {
            id         : 17,
            name       : 'Arrive 5',
            startDate  : '2022-03-23T01:00',
            endDate    : '2022-03-23T03:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 18,
            name       : 'Unload 5',
            startDate  : '2022-03-23T04:00',
            endDate    : '2022-03-23T06:00',
            iconCls    : 'b-fa b-fa-snowplow',
            eventColor : 'orange'
        },
        {
            id         : 19,
            name       : 'Load 5',
            startDate  : '2022-03-23T08:00',
            endDate    : '2022-03-23T11:00',
            iconCls    : 'b-fa b-fa-truck-loading',
            eventColor : 'green'
        },
        {
            id         : 20,
            name       : 'Depart 5',
            startDate  : '2022-03-23T14:00',
            endDate    : '2022-03-23T16:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 21,
            name       : 'Arrive 6',
            startDate  : '2022-03-23T02:00',
            endDate    : '2022-03-23T07:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        },
        {
            id         : 22,
            name       : 'Reload 6',
            startDate  : '2022-03-23T10:00',
            endDate    : '2022-03-23T16:00',
            iconCls    : 'b-fa b-fa-snowplow',
            eventColor : 'orange'
        },
        {
            id         : 23,
            name       : 'Depart 6',
            startDate  : '2022-03-23T19:00',
            endDate    : '2022-03-23T23:00',
            iconCls    : 'b-fa b-fa-arrow-right',
            eventColor : 'blue'
        }
    ],
    assignments : [
        { id : 1, resourceId : 1, eventId : 1 },
        { id : 2, resourceId : 1, eventId : 2 },
        { id : 3, resourceId : 4, eventId : 3 },
        { id : 4, resourceId : 4, eventId : 4 },
        { id : 5, resourceId : 7, eventId : 5 },
        { id : 6, resourceId : 7, eventId : 6 },
        { id : 7, resourceId : 8, eventId : 7 },
        { id : 8, resourceId : 8, eventId : 8 },
        { id : 9, resourceId : 14, eventId : 9 },
        { id : 10, resourceId : 14, eventId : 10 },
        { id : 11, resourceId : 13, eventId : 11 },
        { id : 12, resourceId : 13, eventId : 12 },
        { id : 13, resourceId : 10, eventId : 13 },
        { id : 14, resourceId : 10, eventId : 14 },
        { id : 15, resourceId : 10, eventId : 15 },
        { id : 16, resourceId : 2, eventId : 16 },
        { id : 17, resourceId : 3, eventId : 16 },
        { id : 18, resourceId : 6, eventId : 16 },
        { id : 19, resourceId : 2, eventId : 17 },
        { id : 20, resourceId : 2, eventId : 18 },
        { id : 21, resourceId : 5, eventId : 19 },
        { id : 22, resourceId : 5, eventId : 20 },
        { id : 23, resourceId : 11, eventId : 21 },
        { id : 24, resourceId : 11, eventId : 22 },
        { id : 25, resourceId : 11, eventId : 23 }
    ],
    dependencies : [
        { id : 1, from : 1, to : 2, lag : 1, lagUnit : 'hour' },
        { id : 2, from : 2, to : 3, lag : 1, lagUnit : 'hour' },
        { id : 3, from : 3, to : 4, lag : 1, lagUnit : 'hour' },
        { id : 4, from : 5, to : 6, lag : 1, lagUnit : 'hour' },
        { id : 5, from : 6, to : 7, lag : 1, lagUnit : 'hour' },
        { id : 6, from : 7, to : 8, lag : 1, lagUnit : 'hour' },
        { id : 7, from : 9, to : 10, lag : 1, lagUnit : 'hour' },
        { id : 8, from : 10, to : 11, lag : 1, lagUnit : 'hour' },
        { id : 9, from : 11, to : 12, lag : 1, lagUnit : 'hour' },
        { id : 10, from : 13, to : 14, lag : 2, lagUnit : 'hour' },
        { id : 11, from : 14, to : 15, lag : 2, lagUnit : 'hour' },
        { id : 12, from : 17, to : 18, lag : 1, lagUnit : 'hour' },
        { id : 13, from : 18, to : 19, lag : 2, lagUnit : 'hour' },
        { id : 14, from : 19, to : 20, lag : 3, lagUnit : 'hour' },
        { id : 15, from : 21, to : 22, lag : 3, lagUnit : 'hour' },
        { id : 16, from : 22, to : 23, lag : 3, lagUnit : 'hour' }
    ]

};

const
    schedulerPro = new SchedulerPro({
        appendTo   : 'container',
        startDate  : '2022-03-23',
        endDate    : '2022-03-24',
        viewPreset : 'hourAndDay',
        forceFit   : true,
        columns    : [
            {
                type           : 'resourceInfo',
                text           : 'Name',
                field          : 'name',
                showEventCount : false,
                width          : 150
            }
        ],

        resources    : data.resources,
        events       : data.events,
        assignments  : data.assignments,
        dependencies : data.dependencies
    });

schedulerPro.eventStore.on("change", event => {
    if (event.action === "dataset") {
        event.records.forEach(
            eventRecord => {
                console.log('eventRecord.assignments.length: ', eventRecord.assignments.length);
            }
        );
        setTimeout(() => {
            console.log("after timeout");
            event.records.forEach(
                eventRecord => {
                    console.log('eventRecord.assignments.length: ', eventRecord.assignments.length);
                }
            );
        });
    }
});

const loadData = async () => {
    await schedulerPro.project.loadInlineData({
        eventsData: data.events,
        resourcesData: data.resources,
        assignmentsData: data.assignments,
    });
}

loadData();
```

<img width="263" alt="image" src="https://user-images.githubusercontent.com/21220597/165496556-df71db14-3c1c-4127-8f03-1d7e3a36ddb5.png">

**Comment by pmiklashevych** [comment]
@majewski-merix please shorten your example and start and new forum thread instead. It's hard to follow your conversation here. Thanks

**Comment by chuckn0rris** [comment]
Copied my reply from [here](https://www.bryntum.com/forum/viewtopic.php?f=51&t=21362&p=105760&hilit=4452#p105760)

`EventStore`'s change event triggered when data in that store is changed and you will be able to get data that you loaded into that store. But `SchedulerPro`'s `project` needs to be re-calculated to have all links/dates/assignments updated. Also stores populates one by one, so it's possible that assignments store has not been populated yet, but events store's data is there.

To get changes that required full project calculation, we have [Scheduler.model.ProjectModel#event-dataReady](https://bryntum.com/docs/scheduler-pro/api/Scheduler/model/ProjectModel#event-dataReady)
> Fired when the engine has finished its calculations and the results has been written back to the records.
